common/logisticspipes/gui/guidebook/GuiGuideBook.java

```diff
@@ -0,0 +1,779 @@
+		if (bookItemStack.hasTagCompound()) {
+			state = page;
+			NBTTagCompound nbtTagCompound = bookItemStack.getTagCompound();
+			currentPage.page = gbc.getPage(nbtTagCompound.getInteger("division"), nbtTagCompound.getInteger("chapter"), nbtTagCompound.getInteger("page"));
+			currentPage.setProgress(nbtTagCompound.getFloat("sliderProgress"));
+		} else {
+			state = menu;
+		}
+		loadedNBT = true;
+	}
+
+	@Override
+	public void drawScreen(int mouseX, int mouseY, float partialTicks) {
+		this.mouseX = mouseX;
+		this.mouseY = mouseY;
+		this.drawTransparentOverlay();
+		this.drawCurrentEvent();
+		slider.enabled = area$currentlyDrawnY > area$acrossY;
+		super.drawScreen(mouseX, mouseY, partialTicks);
+		for (GuiGuideBookTabButton tab : tabList) tab.drawButton(mc, mouseX, mouseY, partialTicks);
+		this.drawGuiScroll();
+		if (prevPage.visible && nextPage.visible) this.drawCenteredArrowUnderlay(area$offsetCenterX * 2);
+		this.drawTitle();
+	}
+
+	@Override
+	public void initGui() {
+		this.tabList.clear();
+		this.getDataFromNBT();
+		this.calculateConstraints();
+		this.updateTitle();
+		this.slider = this.addButton(new GuiGuideBookSlider(0, gui$sliderX, gui$sliderY0, gui$sliderY1, z$titleButtons, currentPage.getProgress(), gui$sliderWidth, gui$sliderHeight));
+		this.slider.enabled = false;
+		this.home = this.addButton(new GuiGuideBookTexturedButton(1, gui$x3 - gui$tagWidth, gui$y0 - gui$tagHeight, gui$tagWidth, gui$fullTagHeight, 16, 64, z$titleButtons, 128, 0, 16, 16, false, GuiGuideBookTexturedButton.EnumButtonType.TAB));
+		this.prevPage = this.addButton(new GuiGuideBookTexturedButton(2, area$xCenter - area$offsetCenterX, gui$y3 - 12, gui$arrowWidth, gui$arrowHeight, 0, 0, z$titleButtons, 144, 0, 24, 16, true, GuiGuideBookTexturedButton.EnumButtonType.NORMAL));
+		this.nextPage = this.addButton(new GuiGuideBookTexturedButton(3, area$xCenter + area$offsetCenterX - gui$arrowWidth, gui$y3 - 12, gui$arrowWidth, gui$arrowHeight, 0, 0, z$titleButtons, 168, 0, 24, 16, true, GuiGuideBookTexturedButton.EnumButtonType.NORMAL));
+		for (int i = 0; i < maxTabs; i++) {
+			tabList.add(new GuiGuideBookTabButton(i, gui$x3 - 2 - 2 * gui$tagWidth - (i * gui$tagWidth), gui$y0, gui$tagWidth, gui$tagHeight, ""));
+		}
+		this.updateButtonVisibility();
+	}
+
+	@Override
+	public boolean doesGuiPauseGame() {
+		return false;
+	}
+
+	@Override
+	public void onGuiClosed() {
+		currentPage.setProgress(slider.getProgress());
+		ItemGuideBook.setCurrentPage(currentPage, hand);
+		super.onGuiClosed();
+	}
+
+	@Override
+	protected void actionPerformed(GuiButton button) throws IOException {
+		switch (button.id) {
+			case 1:
+				state = menu;
+				slider.reset();
+				break;
+			case 2:
+				prevPage();
+				break;
+			case 3:
+				nextPage();
+				break;
+			default:
+				break;
+		}
+		updateTitle();
+		updateButtonVisibility();
+	}
+
+	@Override
+	protected void mouseClicked(int mouseX, int mouseY, int mouseButton) throws IOException {
+		if (state == menu)
```

Comment:
leaving away curly brackets is okay with me for one-liners with good visibility, but this certainly isn't the case here.